src/lib/components/Head/Head.svelte

```diff
@@ -7,63 +7,64 @@
 		/** the HTML attributes to attach to the meta tag */
 		attributes?: Record<string, string> | null | undefined;
 	}
-
 	export interface SeoTitleTag {
 		tag: 'title';
 		content: string | null;
 		attributes?: null;
 	}
-
 	export interface RegularMetaAttributes {
 		name: string;
 		content: string;
 	}
-
 	export interface OgMetaAttributes {
 		property: string;
 		content: string;
 	}
-
 	export interface SeoMetaTag {
 		tag: 'meta';
 		content?: null;
 		attributes: RegularMetaAttributes | OgMetaAttributes;
 	}
-
 	export interface FaviconAttributes {
 		sizes: string;
 		type: string;
 		rel: string;
 		href: string;
 	}
-
 	export interface AppleTouchIconAttributes {
 		sizes: string;
 		rel: 'apple-touch-icon';
 		href: string;
 	}
-
 	export interface SeoLinkTag {
 		tag: 'link';
 		content?: null;
 		attributes: FaviconAttributes | AppleTouchIconAttributes;
 	}
-
 	export type SeoTag = SeoTitleTag | SeoMetaTag;
 	export type FaviconTag = SeoMetaTag | SeoLinkTag;
 	export type SeoOrFaviconTag = SeoTag | FaviconTag;
 </script>
 
 <script lang="ts">
 	export let data: Array<TitleMetaLinkTag | SeoOrFaviconTag> = [];
+	// Statically output HTML instead of using conditionals inside <svelte:head>
+	const renderedTags = data
+		.map(({ tag, attributes, content }) => {
+			const attrs = attributes
+				? Object.entries(attributes)
+						.map(([key, value]) => `${key}="${value}"`)
```

Comment:
this is brittle.. I'd feel more safe with a utility like this one that escapes HTML, PLUS some tests that verify that the encoding is working as expected:

```ts
export const tag = (
  tagName: string,
  attrs: Record<string, string | null | undefined>,
  content?: Array<string | null | undefined> | null,
) => {
  const serializedAttrs = [];

  if (attrs) {
    for (const attrName in attrs) {
      if (Object.prototype.hasOwnProperty.call(attrs, attrName)) {
        const value = attrs[attrName];
        if (value) {
          serializedAttrs.push(
            `${escapeString(attrName)}="${escapeString(value)}"`,
          );
        }
      }
    }
  }

  const attrsString =
    serializedAttrs.length > 0 ? ` ${serializedAttrs.join(' ')}` : '';

  return content
    ? `<${tagName}${attrsString}>${content.join('')}</${tagName}>`
    : `<${tagName}${attrsString} />`;
};

// Example usage:

tag('source', {
  srcset: this.data.webpSrcSet,
  sizes: this.sizes ?? this.data.sizes ?? undefined,
  type: 'image/webp',
});
```

Reply:
@stefanoverna good changes, ty! ^

But are we sure this is the even the correct approach to this issue? I'm not familiar enough with Svelte to know if this even the right way to fix this bug. The documentation was really lacking and I found a lot of conflicting info online... nothing else I tried worked, though, except for this.

But it seems strange that you'd need to make a string literal just to avoid a hydration error. I thought Svelte should be pretty mature by now, but if we can't use conditionals in the head... shouldn't a linter rule or such detect that...?

Reply:
Conditionals should work, but they don't and we're not doing anything strange here. We can proceed in parallel with an issue on their end, and in the meantime provide this as an hot fix to avoid problems for our people :)

Reply:
I took a look by copying the Head component. There is nothing with the conditional itself. i.e. content being null, undefined, '' etc, works as expected in svelte as a test and also your component is also doing what you would expect. 

The issue is i believe something to do with how <svelte:element> is behaving and there is some sort of weird behaviour that occurs when it trys to work out if it should close the tag or not. 

Try this, if you have both conditions running, then the meta tags don't close correctly. I.e. the No content condition renders are <meta></meta> 

If however you comment out the with content line, <!--    <svelte:element this={tag} {...attributes}>c{content}</svelte:element>--> then the No content will close correctly. 

Its like, svelte element isn't treating each tag independently, if even one element doesn't close, i.e. say you have a title tag. then none of the close correctly.

I evevn tried using snippets and seperate components. So this may be an issue with svelte/element and perhaps the meta tag and client side hydration perhaps? Its beyond me at this point.

Reply:
@stefanoverna Added some tests and cleaned it up a bit. Big change at https://github.com/datocms/datocms-svelte/pull/14/files?diff=split&w=1#r1876437362 and minor ones throughout.

@emileswain, I think we should open a new issue with Svelte directly with a minimal example and discuss the `<svelte:element>` stuff there. I'll open an issue and tag you there. TY again for the help here!